### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by distantnative**
## This PR …

I checked composer and npm upgrades release notes individually.
You probably need to update your local setups to match the cs fixer version and similar.

### Enhancements
- Upgraded dependencies

## Ready?

- [x] ~~Unit tests for fixed bug/feature~~
- [x] ~~In-code documentation (wherever needed)~~
- [x] Tests and checks all pass

### For review team

- [x] Add changes to release notes draft in Notion
- [x] ~~Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)~~

### CODE REVIEW

**Comment by distantnative** [comment]
@lukasbestle Do you know why the composer stuff in the CI does not install now?

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -174,7 +174,7 @@ jobs:
           extensions: ${{ env.extensions }}
           coverage: none
           tools: |
-            composer:2.3.7, composer-normalize:2.28.0, composer-require-checker:4.0.0,
+            composer:2.4.1, composer-normalize:2.28.3, composer-require-checker:4.1.0,
```
Comment (afbora) [inline .github/workflows/backend.yml]: I've done [few tests](https://github.com/getkirby/kirby/tree/afbora/test-dependencies). `composer-normalize` doesn't support composer versions newer than `2.3.8`.

Actually it seems to support 2.3.9 https://github.com/ergebnis/composer-normalize/pull/942 but I guess there is another problem

Comment (distantnative) [reply .github/workflows/backend.yml]: thanks! downgraded for now

File: src/Cache/FileCache.php [hunk 2]
```diff
@@ -79,19 +79,19 @@ protected function file(string $key): string
 				case '/':
 					break;
 
-				// backslashes get their own marker in the path
-				// to differentiate the cache key from one with forward slashes
+					// backslashes get their own marker in the path
+					// to differentiate the cache key from one with forward slashes
```
Comment (lukasbestle) [inline src/Cache/FileCache.php:84]: We need to move these comments below the `case` statements. With the new indentation they are quite confusing.

File: src/Cms/Email.php [hunk 3]
```diff
@@ -123,7 +122,7 @@ protected function template(): void
 					$this->props['body']['text'] = $text->render($data);
 				}
 
-				// fallback to single email text template
+			// fallback to single email text template
```
Comment (lukasbestle) [inline src/Cms/Email.php:125]: Same here

File: src/Database/Sql.php [hunk 4]
```diff
@@ -708,11 +707,11 @@ public function splitIdentifier($table, $identifier): array
 			case 1:
 				return [$table, $this->unquoteIdentifier($parts[0])];
 
-			// qualified identifier
+				// qualified identifier
```
Comment (lukasbestle) [inline src/Database/Sql.php]: And here

File: src/Toolkit/Html.php [hunk 5]
```diff
@@ -594,7 +594,7 @@ public static function youtube(string $url, array $options = [], array $attr = [
 
 				break;
 
-			// regular video URLs
+				// regular video URLs
```
Comment (lukasbestle) [inline src/Toolkit/Html.php]: And here

**Review by lukasbestle — COMMENTED** [review]
Wondering why the frontend and integration workflows are not executed on the `push` event. Because the `package.json` was changed, they should have.

File: src/Cache/FileCache.php (hunk 2 above)
Comment (afbora) [reply src/Cache/FileCache.php:84]: To be honest, these indentations are getting on my nerves 😩

Comment (lukasbestle) [reply src/Cache/FileCache.php:84]: We only need to fix them once by moving the comments, then we should be fine in the long term.

File: src/Cms/Email.php (hunk 3 above)
Comment (distantnative) [reply src/Cms/Email.php:125]: I think this one makes actually sense as is

**Comment by distantnative** [comment]
@lukasbestle Interestingly, from the frontend workflow file the code style runs, the unit tests get skipped. SO it's probably not related to the `on:` config of the frontend file. Maybe it's this part:

```
    if: >
      github.event_name != 'pull_request' ||
      github.event.pull_request.head.repo.full_name != github.repository
```

? We have the same in the E2E tests, so if that if statement is somehow flawed, could be that.

**Comment by lukasbestle** [comment]
We also have the same in the `backend` workflow file. It shouldn't prevent the action from being run on `push` though as on `push` the event name is `!= 'pull_request'`.

File: src/Cms/Email.php (hunk 3 above)
Comment (lukasbestle) [reply src/Cms/Email.php:125]: Ah, right. Missed that this one was changed in the opposite direction. 👍

**Comment by distantnative** [comment]
Though if I read it correctly, all pull_request ones are skipped for backend as well, only the push ones did run for backend.

**Comment by lukasbestle** [comment]
Yes, that's what that `if:` line does. For backend it works, but for frontend it doesn't. Frontend doesn't even run on `push`.

**Review by lukasbestle — APPROVED** [review]
I think we can still merge this PR already. I think the frontend workflow issue is independent from the changes in this PR. It only surfaced here.

**Pull request merged**